drivers/i3c/i3c_mcux.c

```diff
@@ -1124,6 +1108,17 @@ static int mcux_i3c_transfer(const struct device *dev,
 			}
 		}
 
+		/* Broadcast 7h7e first unless told to skip broadcast header */
+		if ((i == 0) && ((msgs[i].flags & I3C_MSG_NBCH) == 0)) {
```

Comment:
If the previous message is a STOP message, then the next message won't have the broadcast header where it is probably intended to have it unless if it has that flag.

Reply:
I updated the patch.